lib/coverband/adapters/s3_report_writer.rb

```diff
@@ -0,0 +1,59 @@
+# frozen_string_literal: true
+
+module Coverband
+  module Adapters
+    ###
+    # TODO: this is currently a html report writer
+    # this should support writing coverage the current method should be fine
+    # to write every report to S3 and sum them later or use the 2 pass
+    # method we do for redis if in a background thread
+    ###
+    class S3ReportWriter
+      def initialize(bucket_name, options = {})
+        @bucket_name = bucket_name
+        @region = options[:region]
+        @access_key_id = options[:access_key_id]
+        @secret_access_key = options[:secret_access_key]
+        begin
+          require 'aws-sdk'
+        rescue StandardError
+          err_msg = 'coverband requires aws-sdk in order use S3ReportWriter.'
+          Coverband.configuration.logger.error err_msg
+          return
+        end
+      end
+
+      def persist!
+        object.put(body: coverage_content)
+      end
+
+      private
+
+      def coverage_content
+        version = Gem::Specification.find_by_name('simplecov-html').version.version
+        File.read("#{SimpleCov.coverage_dir}/index.html").gsub("./assets/#{version}/", '')
+      rescue StandardError
```

Comment:
Why do we need a rescue here?

Reply:
cool call let me fix

Reply:
I was going to fix this but unfortunately, I can't remember the case and condition that caused this fallback. I think I will have to leave it until I can figure out a way to recreate the issue.